infra/cifuzz/cifuzz.py

```diff
@@ -320,24 +323,70 @@ def get_files_covered_by_target(latest_cov_info, target_name,
     return None
 
   # Cases like curl there is /src/curl and /src/curl_fuzzers/ are handled.
-  if not oss_fuzz_project_base.endswith('/'):
-    oss_fuzz_project_base += '/'
+  if not src_in_docker.endswith('/'):
+    src_in_docker += '/'
 
   affected_file_list = []
   for file in coverage_per_file:
-    if not file['filename'].startswith(oss_fuzz_project_base):
+    if not file['filename'].startswith(src_in_docker):
       continue
     if not file['summary']['regions']['count']:
       # Don't consider a file affected if code in it is never executed.
       continue
 
-    relative_path = file['filename'].replace(oss_fuzz_project_base, '')
+    relative_path = file['filename'].replace(src_in_docker, '')
     affected_file_list.append(relative_path)
   if not affected_file_list:
     return None
   return affected_file_list
 
 
+def keep_affected_fuzzers(project_name, out_dir, files_changed, src_in_docker):
+  """Keeps only the affected fuzzers in the out directory.
+
+  Args:
+    project_name: The name of the relevant OSS-Fuzz project.
+    out_dir: The location of the fuzzer binaries.
+    files_changed: A list of files changed from origin.
+    src_in_docker: The location of the source dir in the docker image.
+  """
+  if not files_changed:
+    logging.info('No files changed from origin.')
+    return
+  fuzzer_paths = utils.get_fuzz_targets(out_dir)
+  if not fuzzer_paths:
+    logging.error('No fuzzers found in out dir.')
+    return
+
+  latest_cov_report_info = get_latest_cov_report_info(project_name)
+  if not latest_cov_report_info:
+    logging.error('Could not download latest coverage report.')
+    return
+  affected_fuzzers = []
+  for fuzzer in fuzzer_paths:
+    covered_files = get_files_covered_by_target(latest_cov_report_info,
+                                                os.path.basename(fuzzer),
+                                                src_in_docker)
+
+    if covered_files and set(covered_files) & set(files_changed):
+      affected_fuzzers.append(os.path.basename(fuzzer))
+
+  if not affected_fuzzers:
+    logging.info('Not using affected fuzzers.')
```

Comment:
This message seems a bit confusing to me. Is this because not having affected fuzzers indicates there is something wrong with coverage and we should bail?  If so, let's make the message more clear e.g. 'No affected fuzzers detected, keeping all as fallback'. or something similar.

Reply:
+1, the message isn't very descriptive

Reply:
Changed to Oliver's suggestion.